Add Dec Monthly (#112)
tikv/website: content/blog/monthly-december-2019.md

---
title: This Month in TiKV - December 2019
date: 2020-1-7
author: Ana Hobden
---

<!-- Fill in the below from the API: https://developer.github.com/v4/explorer/ -->

> Happy new year, new decade! How did our contributors wrap up the decade? Let's take a look, then take a peek at what's coming this year!

Reviewer: ```suggestion
Happy new year, and happy new decade! How did our contributors wrap up the decade? Let's take a look, and then take a peek at what's coming this year!
```

## Releases

> This month our team made 4 TiKV releases! One doesn't *really* count, since it's a beta-release, but we'll count it anyways! 3.0.x and 2.1.x releases are minor releases, and 3.1.0-beta is our upcoming next major release before 4.0.

Reviewer: ```suggestion
This month our team made 4 TiKV releases - 3.0.x and 2.1.x releases are minor releases, and 3.1.0-beta is our upcoming next major release before 4.0.
```

You can review the changelogs here:

* [3.0.7](https://github.com/tikv/tikv/releases/tag/v3.0.7)
* [3.0.8](https://github.com/tikv/tikv/releases/tag/v3.0.8)
* [2.1.19](https://github.com/tikv/tikv/releases/tag/v2.1.19)
* [3.1.0-beta](https://github.com/tikv/tikv/releases/tag/v3.1.0-beta)

Upgrading? Here's things you should know:

> * In **3.0.8** we reduced the severity of errors occuring in the coprocessor to `warn`.

Author: As far as I can understand, no. You can see https://github.com/tikv/tikv/pull/6051

* In **3.0.8** the default value of `split-region-on-table` is now `false` to help reduce region creation pressure during table creation/deletion.
* In **3.0.8** we now welcome you to experiment with a new feature to get flame graphs from a running TiKV. Try visiting `$SERVER:20180/debug/pprof/profile?seconds=10` to see a flamegraph of that TiKV. We'll be adding more details to our documentation about this soon!
* Version **3.1.0** is a beta and should not be adopted in production deployments. It includes new follower read and backup/restore features.

## News

TiKV 3.1.0 is now in beta, with a stable release coming soon! This includes new major features like follower read and distributed backup/restore.

> Follower read will allow clients to read data from followers in a region, reducing pressure on the region leader. Distributed backup and restore are a long awaited feature to make it more practical and efficient to both prepare for the worst, and handle when it does happen.

Reviewer 2: ```suggestion
Follower read will allow clients to read data from followers in a region, reducing pressure on the region leader. Distributed backup and restore are a long awaited feature to make it more practical and efficient to both prepare for the worst, and handle it if it does happen.
```

Author: You are such an optimist!

> Follower read will allow clients to read data from followers in a region, reducing pressure on the region leader. Distributed backup and restore are a long awaited feature to make it more practical and efficient to both prepare for the worst, and handle it if it does happen.

Reviewer: ```suggestion
Follower read will allow clients to read data from followers in a region, reducing pressure on the region leader. Distributed backup and restore is a long-awaited feature to make TiKV more practical and efficient to both prepare for the worst, and handle it if it does happen.
```

## Reading materials

Tang Liu wrote an in depth article about [how TiKV handles read and write operations](https://tikv.org/blog/how-tikv-reads-writes/) over on the TiKV blog.

> Yuanli Wang wrote about recent work on [AutoTiKV](https://pingcap.com/blog/autotikv-tikv-tuning-made-easy-by-machine-learning/), a tuning tool for DBAs to help them find the ideal workload settings for their TiKVs.

Reviewer: ```suggestion
Yuanli Wang wrote about recent work on [AutoTiKV](https://pingcap.com/blog/autotikv-tikv-tuning-made-easy-by-machine-learning/), an AI-powered tuning tool for DBAs to help them find the ideal workload settings for their TiKVs.
```

You can review the changelogs here:

* [3.0.7](https://github.com/tikv/tikv/releases/tag/v3.0.7)
* [3.0.8](https://github.com/tikv/tikv/releases/tag/v3.0.8)
* [2.1.19](https://github.com/tikv/tikv/releases/tag/v2.1.19)
* [3.1.0-beta](https://github.com/tikv/tikv/releases/tag/v3.1.0-beta)

Upgrading? Here's things you should know:

* In **3.0.8** we reduced the severity of errors occuring in the coprocessor to `warn`.
* In **3.0.8** the default value of `split-region-on-table` is now `false` to help reduce region creation pressure during table creation/deletion.
* In **3.0.8** we now welcome you to experiment with a new feature to get flame graphs from a running TiKV. Try visiting `$SERVER:20180/debug/pprof/profile?seconds=10` to see a flamegraph of that TiKV. We'll be adding more details to our documentation about this soon!
* Version **3.1.0** is a beta and should not be adopted in production deployments. Please do try it out and let us know if you find issues. It includes new follower read and backup/restore features.

## News

TiKV 3.1.0 is now in beta, with a stable release coming soon! This includes new major features like follower read and distributed backup/restore.

> <<<<<<< HEAD

Reviewer: There are some conflicting texts here. Otherwise, it looks good to me.

Author: Whoops
